app/scripts/modules/core/src/managed/ManagedResourceObject.tsx

```diff
@@ -20,13 +23,20 @@ function getIconTypeFromKind(kind: string): string {
 export const ManagedResourceObject = ({
   resource: {
     kind,
-    artifact,
     moniker: { app, stack, detail },
   },
-}: IManagedResourceObjectProps) => (
-  <ObjectRow
-    icon={getIconTypeFromKind(kind)}
-    title={[app, stack, detail].filter(Boolean).join('-')}
-    metadata={artifact?.versions?.current || 'unknown version'}
-  />
-);
+  artifact,
+}: IManagedResourceObjectProps) => {
+  const { version: currentVersion, buildNumber: currentBuild } = parseName(artifact?.versions.current || '') || {};
+  return (
+    <ObjectRow
+      icon={getIconTypeFromKind(kind)}
+      title={[app, stack, detail].filter(Boolean).join('-')}
+      metadata={
+        artifact?.versions.current && (
```

Comment:
I don't really know if I love any of the way this is structured/written, but it's a start 🤷‍♂